Fix phpdoc especially return types

dbellettini opened the pull request:
We used phpstan to get warning about types.

Thanks
Davide

jszobody commented:
Thanks for this. Two issues:

**Changing return types**
I'm not sure I like changing all the `@return $this` to `@return self`. According to the proposed (now abandoned) PSR-5 `$this` is a stricter form of `self` that ensure it is the exact class (not a parent or subclass) being returned.

I went to the phpstan playground and create a simple method:

```php
/**
 * @return $this
 */
public function foo() {
	return $this;
}
```

PHPStan said this was great, no errors. Curious why in your case PHPStan wants to change them?

**Moved line**
[line 233](https://github.com/stechstudio/backoff/pull/7/files#diff-ec9872e88333d7c634a8905058bd5621R233) was moved outside the loop, that needs to be put back.

dbellettini committed:
Fix phpdoc especially return types

We used phpstan to get warning about types

dbellettini commented:
Hello @jszobody the `$this` thing was a mistake. I reverted it
I also reverted line 233.
Please let me know if this new version looks good to you

Cheers
Davide

jszobody commented:
Thanks! Merged and tagged a 1.0.1 release
